### ISSUE TITLE
Implement SAME_UPPER auto padding for Maxpool in Gorgonnx

### ISSUE THREAD

**Issue by owulveryck** [owulveryck/onnx-go#74]
The `auto_pad` parameter is deprecated but still used by some model;

This is mandatory for running Tiny yolo v2;
a Naive implementation has been made but does not work (see issue #73 )

The goal of this issue is to follow the implementation of the auto_padding `SAME_UPPER`

This issue is closed when those tests pass:

```
go test -run="ONNX/TestMaxpool.*Upper" -v
=== RUN   TestONNX
=== RUN   TestONNX/TestMaxpool2dPrecomputedSameUpper
=== RUN   TestONNX/TestMaxpool2dSameUpper
--- FAIL: TestONNX (0.01s)
    --- FAIL: TestONNX/TestMaxpool2dPrecomputedSameUpper (0.00s)
        test_structure.go:78:
                Error Trace:    test_structure.go:135
                Error:          Max difference between 1 and 7 allowed is 1e-06, but difference was -6
                Messages:       the two tensors should be equal.
    --- FAIL: TestONNX/TestMaxpool2dSameUpper (0.00s)
        test_structure.go:78:
                Error Trace:    test_structure.go:135
                Error:          Max difference between 1.7640524 and 0.978738 allowed is 1e-06, but difference was 0.7853143811225891
                Messages:       the two tensors should be equal.
FAIL
exit status 1
FAIL    github.com/owulveryck/onnx-go/backend/x/gorgonnx        0.028s
```

WIP in branch [maxpool-sameupper](https://github.com/owulveryck/onnx-go/tree/maxpool-sameupper)

**Comment by owulveryck** [comment]
There will be a problem for some models because Gorgonia does not support asymmetric padding;

asymmetric padding is mandatory in some cases to guarantee that the output shape is the same as the input shape.
Don't know how to fix it.

The point is that the `auto padding` is deprecated in onnx, but a lot of backend/models are still using it.

**Comment by owulveryck** [comment]
Related issue [#167 from onnx/models](https://github.com/onnx/models/issues/167)

**Comment by owulveryck** [comment]
Should be fixed with PR #80 and [PR #295 from Gorgonia](https://github.com/gorgonia/gorgonia/pull/295)
